src/sentry/static/sentry/app/views/settings/incidentRules/triggers/actionsPanel/index.tsx

```diff
@@ -54,47 +53,61 @@ type Props = {
   onChange: (triggerIndex: number, triggers: Trigger[], actions: Action[]) => void;
 };
 
-const getPlaceholderForType = (type: ActionType) => {
-  switch (type) {
-    case ActionType.SLACK:
-      return '@username or #channel';
-    case ActionType.MSTEAMS:
-      //no prefixes for msteams
-      return 'username or channel';
-    case ActionType.PAGERDUTY:
-      return 'service';
-    default:
-      throw Error('Not implemented');
-  }
+/**
+ * When a new action is added, all of it's settings should be set to their default values.
+ * @param actionConfig
+ */
+const getCleanAction = (actionConfig): Action => {
+  return {
+    type: actionConfig.type,
+    targetType:
+      actionConfig &&
+      actionConfig.allowedTargetTypes &&
+      actionConfig.allowedTargetTypes.length > 0
+        ? actionConfig.allowedTargetTypes[0]
+        : null,
+    targetIdentifier: '',
+    integrationId: actionConfig.integrationId,
+    options: actionConfig.options || null,
+  };
 };
+
 /**
- * Lists saved actions as well as control to add a new action
+ * Actions have a type (e.g. email, slack, etc), but only some have
+ * an integrationId (e.g. email is null). This helper creates a unique
+ * id based on the type and integrationId so that we know what action
+ * a user's saved action corresponds to.
  */
-class ActionsPanel extends React.PureComponent<Props> {
-  /**
-   * Actions have a type (e.g. email, slack, etc), but only some have
-   * an integrationId (e.g. email is null). This helper creates a unique
-   * id based on the type and integrationId so that we know what action
-   * a user's saved action corresponds to.
-   */
-  getActionUniqueKey({type, integrationId}: Pick<Action, 'type' | 'integrationId'>) {
-    return `${type}-${integrationId}`;
-  }
+const getActionUniqueKey = ({
+  type,
+  integrationId,
+}: Pick<Action, 'type' | 'integrationId'>) => {
+  return [type, integrationId].filter(x => !!x).join('-');
+};
 
-  /**
-   * Creates a human-friendly display name for the integration based on type and
-   * server provided `integrationName`
-   *
-   * e.g. for slack we show that it is slack and the `integrationName` is the workspace name
-   */
-  getFullActionTitle({
-    type,
-    integrationName,
-  }: Pick<MetricActionTemplate, 'type' | 'integrationName'>) {
-    return `${ActionLabel[type]}${integrationName ? ` - ${integrationName}` : ''}`;
+/**
+ * Creates a human-friendly display name for the integration based on type and
+ * server provided `integrationName`
+ *
+ * e.g. for slack we show that it is slack and the `integrationName` is the workspace name
+ */
+const getFullActionTitle = ({
+  type,
+  integrationName,
+  status,
```

Comment:
I would maybe include in the comment what `status` is for. Integrations have status's as well but they are for things like disabled, pending deletion, etc. which is different than the status of a sentry app